### PR TITLE
Support larger custom route renderers

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
**Description:**
Adjusts color handling to be more consistent and allow for more situations.

LOOKING FOR FEEDBACK:
- things are not quite working: agency id is not showing up in the itinerary response, which is causing the incorrect color to appear in transitive

**PR Checklist:**
- [ ] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [ ] Are all languages supported (Internationalization/Localization)?
- [ ] Are appropriate Typescript types implemented?

### CODE REVIEW

**Review by daniel-heppner-ibigroup — APPROVED** [review]
The route colors in transitive are working for me! Looks good

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Should we keep calculating a settings-based route color as fallback?

File: lib/actions/apiV2.js [hunk 1]
```diff
@@ -700,10 +700,6 @@ export const findRoute = (params) =>
             newRoute.patterns = routePatterns
             // TODO: avoid explicit behavior shift like this
             newRoute.v2 = true
-            newRoute.color = getRouteColorBasedOnSettings(
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply lib/actions/apiV2.js:703]: The reason it is removed here is that we instead do the override in every location where the color is used. This gives us more flexibility in how we use the color. Try it! a feed with no color still gets overriden

**Review by binh-dam-ibigroup — APPROVED** [review]
I think my questions have been answered.

**Pull request merged**